test/src/unit-vfs-read-log-modes.cc

```diff
@@ -0,0 +1,70 @@
+/**
+ * @file unit-vfs-read-log-modes.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Tests for VFS read log modes.
+ */
+
+#include <test/support/tdb_catch.h>
+#include "tiledb/common/logger_public.h"
+#include "tiledb/sm/filesystem/vfs.h"
+#include "tiledb/sm/storage_manager/context.h"
+
+using namespace tiledb::sm;
+
+TEST_CASE("VFS Read Log Modes", "[vfs][read-log-modes]") {
+  std::string mode = GENERATE(
+      "", "fragments", "fragment_files", "all_files", "all_reads", "bad_value");
+
+  LOG_WARN("vfs.read_logging_mode = '" + mode + "'");
+
+  Config cfg;
+  throw_if_not_ok(cfg.set("config.logging_level", "3"));
+  throw_if_not_ok(cfg.set("vfs.read_logging_mode", mode));
```

Comment:
There doesn't seem to be a test case for when this variable isn't set at all, not even to an empty string. I think there might be a possibility for `std::bad_optional_access` in the `VFSParams` constructor.

Reply:
The empty string is covered by the GENERATE. I'll add a case to not set it at all to prove that its been set in the default config parameters.